dataland-backend/src/main/kotlin/org/dataland/datalandbackend/validator/PageRangeString.kt

```diff
@@ -0,0 +1,51 @@
+package org.dataland.datalandbackend.validator
+
+import jakarta.validation.Constraint
+import jakarta.validation.ConstraintValidator
+import jakarta.validation.ConstraintValidatorContext
+import jakarta.validation.Payload
+import kotlin.reflect.KClass
+
+/**
+ * Annotation for validating a page range as a string.
+ */
+@Target(AnnotationTarget.PROPERTY_GETTER)
+@Retention(AnnotationRetention.RUNTIME)
+@Constraint(validatedBy = [PageRangeValidator::class])
+annotation class PageRange(
+    val message: String = "Input validation failed: Invalid page range format.",
+    val groups: Array<KClass<*>> = [],
+    val payload: Array<KClass<out Payload>> = [],
+)
+
+/**
+ * Validator class for validating a page range in the form of "A-B" or a single page number.
+ */
+class PageRangeValidator : ConstraintValidator<PageRange, String> {
+
+    override fun initialize(constraintAnnotation: PageRange) {
+        // No initialization needed
+    }
+
+    override fun isValid(value: String?, context: ConstraintValidatorContext?): Boolean {
+        if (value == null) return true // Consider null as valid
+
+        val regexSinglePage = """^[1-9]\d*$""".toRegex()
+        val regexPageRange = """^([1-9]\d*)-([1-9]\d*)$""".toRegex()
```

Comment:
Compiling the regexes each time they are called seems expensive and unnecessary. I think it would be better to initiate them once per object or even per class and reuse them.